backend/app.py

```diff
@@ -0,0 +1,430 @@
+    if not formData:
+        return jsonify({"result": "Bad request, no data passed to form!"}), 400
+    username = formData['username']
+    password = formData['password']
+    app.logger.info(f"User: {username}")
+
+    creation = User.createUser(username, password, db)
+    if creation[0] is True:
+        return jsonify({"result": "Account creation successful"}), 200
+    elif creation[0] is False and creation[1] == 1:
+        return jsonify({"result": "Username already exists"}), 409
+    else:
+        return jsonify({"result": "Internal service error"}), 500
+
+
+@app.get("/request")
+def buoyRequest():
+    """
+    route for getting buoy data for the station ID passed in the get request.
+    To use this route the following request URL should be utilized:
+    /request?stationID=<string>
+    """
+    db = db_handler
+
+    # get all buoys
+    if request.args.get("stationID") == "all":
+        data = allBuoys(db)
+        if data:
+            return jsonify(data), 200
+        return jsonify({"result": "Error occurred"}), 404
+
+    # get a single buoys data
+    else:
+        param = request.args.get("stationID")
+        if param:
+            data = parseBuoy(param)
+            app.logger.info(data)
+            if data:
+                return jsonify(data), 200
+            return jsonify(
+                            {"error": f"No buoy found for stationID {param}"}
+                           ), 404
+        else:
+            return jsonify({"result": "No station ID passed"}), 400
+
+
+# Buoy data and surf spot routes
+@app.route("/findBuoys")
+def findBuoys():
+    """
+    route to get the nearest buoys to a particular set of coordinates
+    URL to use by frontend:
+    /findBuoys?lat=<float>&long=<float>
+    """
+    db = db_handler
+    lat = request.args.get('lat', type=float)
+    long = request.args.get('long', type=float)
+    coord = (lat, long)
+    if coord[0] and coord[1]:
+        data = haversineCalc(coord, db)
+        return jsonify(data), 200
+    return jsonify({"result": "No coorindates passed"}), 400
+
+
+# ------------------------------------------------------------ #
+# --------------------- PROTECTED ROUTES --------------------- #
+# ------------------------------------------------------------ #
+
+
+@app.route("/auth", methods=["GET", "OPTIONS"])
+@login_required
+def auth():
+    """
+    Used by the frontend to quickly verify if the user is logged in when
+    navigating between pages. It will either return an error if the user
+    is not logged in or it simply returns an empty string if the
+    user is logged in
+    """
+    return ""
+
+
+@app.route("/logout")
+@login_required
+def logout():
+    """
+    Function used to logout a user from a Flask-Login session
+    """
+    logout_user()
+    app.logger.info("User logged out successfully")
+    return jsonify({"result": "Logout Successful"}), 200
+
+
+@app.route("/surfSpot", methods=["GET", "PUT", "POST", "DELETE"])
```

Comment:
normally for urls you should use hyphen-case https://stackoverflow.com/questions/10302179/hyphen-underscore-or-camelcase-as-word-delimiter-in-uris

```suggestion
@app.route("/surf-spot", methods=["GET", "PUT", "POST", "DELETE"])
```